src/Providers/ProviderRegistry.php

```diff
@@ -113,6 +113,18 @@ public function registerProvider(string $className): void
         $this->registeredClassNames[$className] = true;
     }
 
+    /**
+     * Gets a list of all registered provider IDs.
+     *
+     * @since 0.1.0
+     *
+     * @return list<string> List of registered provider IDs.
```

Comment:
We can make this even clearer, which will help the IDE as well:
```suggestion
     * @return list<class-string<ProviderInterface> List of registered provider IDs.
```